components/devtools/actors/watcher.rs

```diff
@@ -211,16 +221,35 @@ impl Actor for WatcherActor {
         _id: StreamId,
     ) -> Result<ActorMessageStatus, ()> {
         let target = registry.find::<BrowsingContextActor>(&self.browsing_context_actor);
+        let root = registry.find::<RootActor>("root");
         Ok(match msg_type {
             "watchTargets" => {
-                let msg = WatchTargetsReply {
-                    from: self.name(),
-                    type_: "target-available-form".into(),
-                    target: target.encodable(),
-                };
-                let _ = stream.write_json_packet(&msg);
+                // As per logs we either get targetType as "frame" or "worker"
+                let target_type = msg
+                    .get("targetType")
+                    .and_then(|v| v.as_str())
+                    .unwrap_or("frame"); // default to "frame"
 
-                target.frame_update(stream);
+                if target_type == "frame" {
```

Comment:
It might be worth an early return and a `warn!` if the `target_type` is an an unexpected value.